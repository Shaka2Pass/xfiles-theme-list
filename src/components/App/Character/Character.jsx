import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image, status, occupation, portrayedby }) => (
  <section>
    <h2>
      Name:{name}
    </h2>
    <img src={image}></img>
    <ul>Status{status}</ul>
    <ul>Occupation{occupation}</ul>
    <ul>Actor:{portrayedby}</ul>
  </section>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired,
  portrayedby: PropTypes.string.isRequired
};

export default Character;

