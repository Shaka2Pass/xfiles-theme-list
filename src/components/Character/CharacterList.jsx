import React from 'react';
import PropTypes from 'prop-types';

const  CharacterList = ({ name, image, status, occupation, portrayedby }) => (
  <section>
    <p>{name}</p>
    {image && <img src={image} alt="image of xfiles character" />}
    {status && <p>Status:{status}</p>}
    {occupation && <p>Occupation:{occupation}</p>}
    {portrayedby && <p>Actor:{portrayedby}</p>}
  </section>
);

CharacterList.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired,
  portrayedby: PropTypes.string.isRequired, 
};

export default CharacterList;
