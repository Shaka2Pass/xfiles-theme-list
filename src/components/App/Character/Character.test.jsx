import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character.jsx';

describe('a single Character', () => {
  it('matches a snapshot of detail', () => {
    const wrapper = shallow(<Character
      name="Dana Scully"
      image="https://vignette.wikia.nocookie.net/x-files/images/3/31/Absalom.jpg/revision/latest/scale-to-width-down/310?cb=20070708201919"
      status="alive"
      occupation="FBI Agent"
      portrayedby="Gillian Anderson"/>);
    expect(wrapper).toMatchSnapshot();
  });
});



