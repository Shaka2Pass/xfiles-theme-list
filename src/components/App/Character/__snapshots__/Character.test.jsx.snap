// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a single Character matches a snapshot of detail 1`] = `
<section>
  <h2>
    Name:
    Dana Scully
  </h2>
  <img
    src="https://vignette.wikia.nocookie.net/x-files/images/3/31/Absalom.jpg/revision/latest/scale-to-width-down/310?cb=20070708201919"
  />
  <ul>
    Status
    alive
  </ul>
  <ul>
    Occupation
    FBI Agent
  </ul>
  <ul>
    Actor:
    Gillian Anderson
  </ul>
</section>
`;
